leapp/utils/lock.py

```diff
@@ -0,0 +1,96 @@
+import os
+import fcntl
+import logging
+
+from leapp.config import get_config
+from leapp.exceptions import ProcessLockError
+
+
+def leapp_lock(lockfile=None):
+    return ProcessLock(lockfile=lockfile)
+
+
+def _acquire_lock(fd):
+    try:
+        fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+        return True
+    except OSError:
+        return False
+
+
+def _read_pid(fd):
+    return os.read(fd, 20)
+
+
+def _write_pid(fd, pid):
+    os.write(fd, str(pid).encode('utf-8'))
+
+
+def _validate_pid(old_pid):
+    try:
+        old_pid = int(old_pid)
+    except ValueError:
+        return False
+
+    return old_pid
+
+
+def _clear_lock(fd):
+    os.lseek(fd, 0, os.SEEK_SET)
+    os.ftruncate(fd, 0)
+
+
+class ProcessLock(object):
+
+    def __init__(self, lockfile=None):
+        self.log = logging.getLogger('leapp.utils.lock')
+        self.lockfile = lockfile if lockfile else get_config().get('lock', 'path')
+
+        self.fd = None
+
+    def _get_pid_from_lockfile(self):
+        running_pid = _read_pid(self.fd)
+        self.log.debug("_get_pid_from_lockfile: running_pid=%s", running_pid)
+
+        if len(running_pid) == 0:
+            # No pid in lockfile
+            return
+
+        running_pid = _validate_pid(running_pid)
+        if not running_pid:
```

Comment:
Note: a pid could be zero so technically this should have been checking if `running_pid is None`.  (In practice, pid 0 is reserved for the kernel so this wouldn't generate a noticable bug.  But it is good to keep in mind for other times you are writing conditions with variables that are numbers.)

Reply:
Very true! Now that I removed the validation function I actually can do away with the whole check altogether. Thanks! Fixed